[🐛 | Bug]: Cannot build snap

### Parsec version tested on:

master

### Platforms tested on:

Linux

### Bug description:

Cannot build snap, the package fail during the build part `parsec` when it install the node dependencies

### Relevant output:

```shell
:: + npm clean-install
:: npm WARN deprecated eslint-config-standard-with-typescript@39.1.1: Please use eslint-config-love, instead.
:: npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated
:: npm WARN deprecated superagent@8.1.2: Please upgrade to v9.0.0+ as we have fixed a public vulnerability with formidable dependency. Note that v9.0.0+ requires Node.js v14.18.0+. See https://github.com/ladjs/superagent/pull/1800 for insight. This project is supported and maintained by the team at Forward Email @ https://forwardemail.net/
:: npm ERR! code 243
:: npm ERR! git dep preparation failed
:: npm ERR! command /snap/node/8520/bin/node /snap/node/8520/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/root/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit --include=dev --include=peer --include=optional --no-package-lock-only --no-dry-run
:: npm ERR! > megashark-lib@0.0.1 prepare
:: npm ERR! > npm run build
:: npm ERR! npm WARN using --force Recommended protections disabled.
:: npm ERR! npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated
:: npm ERR! npm WARN deprecated eslint-config-standard-with-typescript@39.1.1: Please use eslint-config-love, instead.
:: npm ERR! npm ERR! code 243
:: npm ERR! npm ERR! path /root/.npm/_cacache/tmp/git-clonevGnSB6
:: npm ERR! npm ERR! command failed
:: npm ERR! npm ERR! command sh -c npm run build
:: npm ERR!
:: npm ERR! npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-04-25T22_45_50_942Z-debug-0.log
::
:: npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-04-25T22_45_40_028Z-debug-0.log
```

--- a/.github/workflows/package-client.yml
+++ b/.github/workflows/package-client.yml
@@ -5,7 +5,7 @@ on:
     paths:
       - .github/workflows/package-client.yml
       - client/electron/package.js
-      - client/electron/snap
+      - client/electron/snap/*
   workflow_call:
     inputs:
       version:
@@ -112,9 +112,7 @@ jobs:
     needs: version
     runs-on: ubuntu-22.04
     # Always run the job if `version` job is skipped otherwise only if `version` job was successful.
-    # if: ${{ inputs.version_patch_run_id != '' && always() || success() }}
-    # TODO: snap is temporary disabled
-    if: false
+    if: ${{ inputs.version_patch_run_id != '' && always() || success() }}
     steps:
       - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # pin v4.1.6
         with:

--- a/client/electron/snap/snapcraft.yaml
+++ b/client/electron/snap/snapcraft.yaml
@@ -73,8 +73,32 @@ parts:
       cp -va dist/libparsec/index.node "$CRAFT_PART_INSTALL/libparsec.node"
       cp -va dist/libparsec/index.d.ts "$CRAFT_PART_INSTALL/libparsec.d.ts"
 
+  # We pre-package `megashark-lib` to workaround a permission issue during it's `prepare` step when installed as a 3rd dependencies from `client-vite`.
+  megashark-lib:
+    plugin: nil
+    build-snaps:
+      - node/18/stable
+    source: client
+    build-packages:
+      - git
+      - jq
+    override-build: |
+      set -x
+
+      # Clone megashark-lib locally
+      MEGASHARK_COMMIT_ID=$(jq -r '.packages."node_modules/megashark-lib".resolved' package-lock.json | cut -d '#' -f 2)
+      git clone https://github.com/Scille/megashark-lib.git
+      cd megashark-lib
+      git reset --hard $MEGASHARK_COMMIT_ID
+
+      npm clean-install
+      MEGASHARK_ARCHIVE=$(npm pack | tail -n1)
+
+      cp -v $MEGASHARK_ARCHIVE "$CRAFT_STAGE/megashark-lib.tar.gz"
+
   parsec:
     after:
+      - megashark-lib
       - libparsec
     plugin: nil
     source: client
@@ -89,7 +113,10 @@ parts:
       node --version
       npm --version
 
-      npm clean-install
+      # Patch client package.json to use local pre-packaged megashark-lib
+      sed -i "s;megashark-lib\": \".*\";megashark-lib\": \"file:${CRAFT_STAGE}/megashark-lib.tar.gz\";" package.json
+
+      npm install
       npm run electron:copy
 
       cd electron

